database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -65,14 +65,20 @@
    */
   protected int[] columnTypes;
 
+  /**
+   * Need to cache column names to set fields of the input record on {@link PreparedStatement} in the right order.
+   */
+  protected List<String> columns;
+
   /**
    * Used to construct a DBRecord from a StructuredRecord in the ETL Pipeline
    *
    * @param record the {@link StructuredRecord} to construct the {@link DBRecord} from
    */
-  public DBRecord(StructuredRecord record, int[] columnTypes) {
+  public DBRecord(StructuredRecord record, int[] columnTypes, List<String> columns) {
```

Comment:
it seems odd to have a mix of types where one is an array and the other is a list. Additionally, having two separate arguments makes it possible to construct a record where the types and column names don't match up.

We can avoid these problems, if we pass in an array or list of a new ColumnType class, which just contains a column name and type. Or if there is an existing object that the columnTypes array is derived from, that would work too.

Reply:
Changed to use `ColumnType` class.